### ISSUE TITLE
Special Variable for Current Working Directory

### ISSUE THREAD

**Issue by is-jonreeves** [go-task/task#525]
It would be great to have a "special variable" that reflects the Current Working Directory for its context. I am finding myself trying to use the `sh: echo "$PWD"` approach lots and this has [some issues when running more than once](https://github.com/go-task/task/issues/524). Right now I find myself working around this issue alot having to use a combination of the `list | join and osClean` functions to construct the correct Abs Path for my given context.

### Example
If a "special variable" called `.CWD` was implement then, the following would be possible:
```yaml
version: '3'

tasks:
  example1:
    cmds:
      - echo "{{.CWD}}"    # ---> C:\Users\Jon\Projects\taskfile-test
  example2:
    dir: ../
    cmds:
      - echo "{{.CWD}}"    # ---> C:\Users\Jon\Projects
  example3:
    dir: ../../
    cmds:
      - echo "{{.CWD}}"    # ---> C:\Users\Jon
```

### Workarounds
There are a few ways I've found of doing this today without a built-in special variable, but they feel unnecessary:

#### Approach 1: Dynamic Variables (glitched right now)
Note the need to append pairs of `''` to each `sh:` usage.
```yaml
version: '3'
silent: true

vars:
  CWD: { sh: echo $PWD }

tasks:
  example1:
    cmds:
      - echo "{{.CWD}}"    # ---> C:\Users\Jon\Projects\taskfile-test

  example2:
    dir: ../
    vars:
      CWD: { sh: echo $PWD'' }
    cmds:
      - echo "{{.CWD}}"    # ---> C:\Users\Jon\Projects

  example3:
    dir: ../../
    vars:
      CWD: { sh: echo $PWD'''' }
    cmds:
      - echo "{{.CWD}}"    # ---> C:\Users\Jon
```

#### Approach 2: String Processing
Really Repetitive and wouldn't support non-relative paths.
```yaml
version: '3'
silent: true

vars:
  DIR: './'
  CWD: { sh: pwd }

tasks:
  example1:
    dir: '{{.DIR}}'
    cmds:
      - echo "{{.CWD}}"    # ---> C:\Users\Jon\Projects\taskfile-test

  example2:
    dir: '{{.DIR}}'
    vars:
      DIR: ../
      CWD: '{{list .CWD "/" .DIR | join "" | osClean}}'
    cmds:
      - echo "{{.CWD}}"    # ---> C:\Users\Jon\Projects

  example3:
    dir: '{{.DIR}}'
    vars:
      DIR: ../../
      CWD: '{{list .CWD "/" .DIR | join "" | osClean}}'
    cmds:
      - echo "{{.CWD}}"    # ---> C:\Users\Jon
```

**Comment by SwitchedToGitlab** [comment]
I wholeheartedly agree with this request. I also believe there should be a special variable for `$HOME` and `$USERNAME`. 

@andreynering I'm happy to look into this as a first PR if you're interested in adding the functionality. I appreciate any thoughts you have on the matter.

**Comment by andreynering** [comment]
Hi @is-jonreeves and @SwitchedToGitlab,

Yep, I this this makes sense to have! 👍 

---

I think we don't need variables for username and home, though. These are available as environment variables, and do not change between task runs.

**Comment by andreynering** [comment]
I just realized this is proposed before.

Closed as a duplicate of #215.
